test/state/host.cpp

```diff
@@ -228,9 +228,40 @@ std::optional<evmc_message> Host::prepare_message(evmc_message msg)
             else
             {
                 assert(msg.kind == EVMC_EOFCREATE);
-                const bytes_view initcontainer{msg.code, msg.code_size};
-                msg.recipient =
-                    compute_eofcreate_address(msg.sender, msg.create2_salt, initcontainer);
+
+                // Indicator of an EOF creation tx - EVMC_EOFCREATE at depth 0.
```

Comment:
I'm not a big fan of this. Alternative is to have a next msg.kind like EVMC_EOFCREATIONTX. For starters decided to leverage depth. WDYT?

Reply:
I think it's fine, there're a couple of places above in this same functions with this check, meaning the same.